core/src/test/scala/unit/kafka/server/LogDirFailureTest.scala

```diff
@@ -191,12 +198,27 @@ class LogDirFailureTest extends IntegrationTestHarness {
     TestUtils.pollUntilAtLeastNumRecords(consumer, 1)
 
     // There should be no remaining LogDirEventNotification znode
-    assertTrue(zkClient.getAllLogDirEventNotifications.isEmpty)
+    if (quorum == "zk") {
+      assertTrue(zkClient.getAllLogDirEventNotifications.isEmpty)
+    }
 
-    // The controller should have marked the replica on the original leader as offline
-    val controllerServer = servers.find(_.kafkaController.isActive).get
-    val offlineReplicas = controllerServer.kafkaController.controllerContext.replicasInState(topic, OfflineReplica)
-    assertTrue(offlineReplicas.contains(PartitionAndReplica(new TopicPartition(topic, 0), leaderServerId)))
+    if (quorum == "kraft") {
+      waitUntilTrue(() => {
+        brokers.exists(broker => {
+          val hasOfflineDir = broker.asInstanceOf[BrokerServer].logDirFailureChannel.hasOfflineLogDir(failedLogDir.toPath.toString)
```

Comment:
We can be more specific than this. It shouldn't just be any broker, but the target of `TestUtils.causeLogDirFailure`, which I think is `originalLeaderServer`.

Reply:
Fair point, change made